tcc compatibility

I've written a wrapper for [V/vlang][https://github.com/vlang/v] which works fine with gcc and clang - but now I'm trying to compile Chipmunk2D with (the blazing fast) [tcc](https://repo.or.cz/tinycc.git/) (the default compiler shipped with [V][https://github.com/vlang/v])
```
CC=tcc cmake
-- The C compiler identification is TinyCC
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/tcc
-- Check for working C compiler: /usr/bin/tcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Configuring Chipmunk2D version 7.0.3
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libOpenGL.so   
-- Configuring done
-- Generating done

make
Scanning dependencies of target chipmunk_static
[  1%] Building C object src/CMakeFiles/chipmunk_static.dir/chipmunk.c.o
[  2%] Building C object src/CMakeFiles/chipmunk_static.dir/cpArbiter.c.o
[  3%] Building C object src/CMakeFiles/chipmunk_static.dir/cpArray.c.o
[  4%] Building C object src/CMakeFiles/chipmunk_static.dir/cpBBTree.c.o
[  5%] Building C object src/CMakeFiles/chipmunk_static.dir/cpBody.c.o
[  6%] Building C object src/CMakeFiles/chipmunk_static.dir/cpCollision.c.o
[  7%] Building C object src/CMakeFiles/chipmunk_static.dir/cpConstraint.c.o
[  8%] Building C object src/CMakeFiles/chipmunk_static.dir/cpDampedRotarySpring.c.o
[  9%] Building C object src/CMakeFiles/chipmunk_static.dir/cpDampedSpring.c.o
[ 10%] Building C object src/CMakeFiles/chipmunk_static.dir/cpGearJoint.c.o
[ 11%] Building C object src/CMakeFiles/chipmunk_static.dir/cpGrooveJoint.c.o
[ 12%] Building C object src/CMakeFiles/chipmunk_static.dir/cpHashSet.c.o
[ 13%] Building C object src/CMakeFiles/chipmunk_static.dir/cpHastySpace.c.o
[ 14%] Building C object src/CMakeFiles/chipmunk_static.dir/cpMarch.c.o
[ 15%] Building C object src/CMakeFiles/chipmunk_static.dir/cpPinJoint.c.o
[ 16%] Building C object src/CMakeFiles/chipmunk_static.dir/cpPivotJoint.c.o
[ 17%] Building C object src/CMakeFiles/chipmunk_static.dir/cpPolyShape.c.o
/home/user/vdev/chipmunk-dev/chipmunk-mod/c/Chipmunk2D/src/cpPolyShape.c:240: error: index too large
src/CMakeFiles/chipmunk_static.dir/build.make:446: recipe for target 'src/CMakeFiles/chipmunk_static.dir/cpPolyShape.c.o' failed
```
I'd be super awesome if we could get Chipmunk2D compiling with `tcc` somehow - especially because compile times would improve dramatically

--- a/demo/ChipmunkDebugDraw.c
+++ b/demo/ChipmunkDebugDraw.c
@@ -273,7 +273,7 @@ void ChipmunkDebugDrawPolygon(int count, const cpVect *verts, cpFloat radius, cp
 
 void ChipmunkDebugDrawBB(cpBB bb, cpSpaceDebugColor color)
 {
-	cpVect verts[] = {
+	cpVect verts[4] = {
 		cpv(bb.r, bb.b),
 		cpv(bb.r, bb.t),
 		cpv(bb.l, bb.t),

--- a/demo/Planet.c
+++ b/demo/Planet.c
@@ -62,7 +62,7 @@ add_box(cpSpace *space)
 	const cpFloat size = 10.0f;
 	const cpFloat mass = 1.0f;
 	
-	cpVect verts[] = {
+	cpVect verts[4] = {
 		cpv(-size,-size),
 		cpv(-size, size),
 		cpv( size, size),

--- a/demo/Plink.c
+++ b/demo/Plink.c
@@ -72,7 +72,7 @@ init(void)
 	cpShape *shape;
 	
 	// Vertexes for a triangle shape.
-	cpVect tris[] = {
+	cpVect tris[4] = {
 		cpv(-15,-15),
 		cpv(  0, 10),
 		cpv( 15,-15),

--- a/demo/Pump.c
+++ b/demo/Pump.c
@@ -109,7 +109,7 @@ init(void)
 	cpShapeSetFriction(shape, 0.5f);
 	cpShapeSetFilter(shape, NOT_GRABBABLE_FILTER);
 
-	cpVect verts[] = {
+	cpVect verts[4] = {
 		cpv(-30,-80),
 		cpv(-30, 80),
 		cpv( 30, 64),

--- a/src/cpPolyShape.c
+++ b/src/cpPolyShape.c
@@ -234,7 +234,7 @@ cpBoxShapeInit(cpPolyShape *poly, cpBody *body, cpFloat width, cpFloat height, c
 cpPolyShape *
 cpBoxShapeInit2(cpPolyShape *poly, cpBody *body, cpBB box, cpFloat radius)
 {
-	cpVect verts[] = {
+	cpVect verts[4] = {
 		cpv(box.r, box.b),
 		cpv(box.r, box.t),
 		cpv(box.l, box.t),

